### PR TITLE
User and wallet cleanup

### PR DESCRIPTION
**Pull request by valentinludu**
Use SWR and less states and effects

### CODE REVIEW

File: components/_app/Web3ConnectionManager/Web3ConnectionManager.tsx [hunk 1]
```diff
@@ -8,9 +8,7 @@ const Web3Connection = createContext({
   isWalletSelectorModalOpen: false,
   connectWallet: () => {},
   closeWalletSelectorModal: () => {},
-  triedEager: false,
-  isConnectingIdentity: false,
```
Comment (valentinludu) [inline components/_app/Web3ConnectionManager/Web3ConnectionManager.tsx:12]: setIsConnectingIdentity was not used at all.
isConnectingIdentity was always false.

Below there is a comment about using this only for unstoppable domains, but we don't have that anymore.

File: hooks/useUser.tsx [hunk 2]
```diff
@@ -54,21 +49,12 @@ export function UserProvider({ children }: { children: ReactNode }) {
...
-  }, []);
-
-  useEffect(() => {
-    if (user?.deletedAt) {
-      charmClient.logout().then(() => {
-        window.location.href = window.location.origin;
-      });
-    }
-  }, [user]);
+  const updateUser = (updatedUser: Partial<LoggedInUser>) => refreshUser(updatedUser);
```
Comment (valentinludu) [inline hooks/useUser.tsx:52]: These 2 functions were left here just for legacy. We are using them in a lot of places but in reality we only need 1.

File: hooks/useWeb3Account.tsx [hunk 3]
```diff
@@ -204,7 +209,7 @@ export function Web3AccountProvider({ children }: { children: ReactNode }) {
       const storedWalletSignature = getStoredSignature(account);
       setSignature(account, storedWalletSignature);
     }
-    // Case 3: user is switching wallets
+    // Case 2: user is switching wallets
```
Comment (mattcasey) [inline hooks/useWeb3Account.tsx:212]: this is still case 3 isnt it?

File: hooks/useWeb3Account.tsx [hunk 4]
```diff
@@ -266,29 +262,27 @@ export function Web3AccountProvider({ children }: { children: ReactNode }) {
...
+        {
+          onSuccess: async (updatedUser) => {
+            logoutWallet();
+
+            setLitAuthSignature(null);
+            setLitProvider(null);
+            setStoredAccount(null);
+            updateUser(updatedUser);
+            activeConnector?.disconnect();
+            await mutate((key) => typeof key === 'string' && key.startsWith(`/nfts/${updatedUser?.id}`));
```
Comment (mattcasey) [inline hooks/useWeb3Account.tsx:280]: i'm just curious whats the difference with this callback method vs how we called mutate before?

**Review by mattcasey — APPROVED** [review]
Great cleanup! Thank you

File: hooks/useWeb3Account.tsx (hunk 3 above)
Comment (valentinludu) [reply hooks/useWeb3Account.tsx:212]: yes.

I renamed it back

File: hooks/useWeb3Account.tsx (hunk 4 above)
Comment (valentinludu) [reply hooks/useWeb3Account.tsx:280]: From official docs, this is how we need to mutate the global cache key.
I tried before the old approach and it did not work.

**Pull request merged**